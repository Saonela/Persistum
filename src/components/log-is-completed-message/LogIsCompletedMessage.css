.log-is-completed-message__illustration {
    padding-top: var(--padding-l);
    margin-top: 50px;
    margin-bottom: var(--margin-m);
    border-top: 1px solid var(--color-border);
}

.log-is-completed-message__illustration svg {
    font-size: 100px;
    color: var(--color-success);
}

.log-is-completed-message__congratulations {
    font-size: 20px;
    padding-bottom: var(--padding-l);
    margin: 0 0 50px;
    border-bottom: 1px solid var(--color-border);
}

.message-navigation-options {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    font-size: var(--font-size);
}

.message-navigation-options > div {
    padding: 8px 8px 16px;
}

button.message-navigation-options__button {
    margin-top: var(--margin);
}

button.log-is-completed-message__go-back {
    font-size: 14px;
}

