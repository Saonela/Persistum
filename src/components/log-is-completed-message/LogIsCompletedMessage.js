import React from "react";
import './LogIsCompletedMessage.css'
import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
import Button from "@material-ui/core/Button";
import {Link} from "react-router-dom";
import FormDate from "../form-view/form-date/FormDate";

function LogIsCompletedMessage({date, onBack}) {

    return (
        <div className="log-is-completed-message">
            <FormDate date={date}/>
            <div className="log-is-completed-message__illustration">
                <CheckCircleOutlineIcon/>
            </div>
            <div className="log-is-completed-message__congratulations">
                Your day has been logged !
            </div>
            <div className="message-navigation-options">
                <div className='message-navigation-options__to-form'>
                    <div>If you have missed something</div>
                    <Button className="message-navigation-options__button" variant="outlined" size="small" onClick={() => onBack()}>Back to logging</Button>
                </div>
                <div className='message-navigation-options__to-calendar'>
                    <div>To review overall statistics</div>
                    <Link to="/calendar">
                        <Button className="message-navigation-options__button" variant="outlined" size="small">Go to calendar</Button>
                    </Link>
                </div>
            </div>
        </div>
    );
}

export default LogIsCompletedMessage;
