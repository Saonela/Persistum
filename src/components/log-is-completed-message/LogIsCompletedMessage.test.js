import LogIsCompletedMessage from "./LogIsCompletedMessage";
import {render} from "@testing-library/react";
import {fireEvent, getByText} from "@testing-library/dom";
import React from "react";
import {BrowserRouter as Router} from "react-router-dom";

it('should emit event on go back button click', () => {
    const spy = jest.fn();
    const {container} = render(<Router><LogIsCompletedMessage date={'2020'} onBack={spy}/></Router>);
    const button = getByText(container, 'Back to logging');
    fireEvent.click(button);
    expect(spy).toHaveBeenCalled();
});
