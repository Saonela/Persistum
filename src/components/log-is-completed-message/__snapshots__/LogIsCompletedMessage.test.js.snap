// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="log-is-completed-message"
  >
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <div
          class="form-date"
        >
          <div
            class="form-date__container"
          >
            <p
              class="form-date__day"
            >
              Wednesday 01
            </p>
            <div
              class="inner-block"
            >
              <p
                class="form-date__year"
              >
                2020
              </p>
              <p
                class="form-date__month"
              >
                January
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="log-is-completed-message__congratulations"
    >
      <div
        class="log-is-completed-message__illustration"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 7.58L10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      Your day has been logged !
    </div>
    <div
      class="message-navigation-options"
    >
      <div
        class="message-navigation-options__to-form"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined app-button app-button--outline message-navigation-options__button MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Back to logging
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div>
          If you have missed something
        </div>
      </div>
      <div
        class="message-navigation-options__to-calendar"
      >
        <a
          href="/calendar"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined app-button app-button--outline message-navigation-options__button MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Go to calendar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <div>
          To review overall statistics
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
